src/Compilers/VisualBasic/Portable/Binding/DeclarationInitializerBinder.vb

```diff
@@ -17,19 +18,27 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' </summary>
         Private ReadOnly _symbol As Symbol
 
+        ''' <summary>
+        ''' Backing field for the AdditionalContainingMembers property
+        ''' </summary>
+        Private ReadOnly _additionalSymbols As ImmutableArray(Of Symbol)
+
         ''' <summary> Root syntax node </summary>
         Private ReadOnly _root As VisualBasicSyntaxNode
 
         ''' <summary>
         ''' Initializes a new instance of the <see cref="DeclarationInitializerBinder"/> class.
         ''' </summary>
         ''' <param name="symbol">The field, property or parameter symbol with an initializer or default value.</param>
+        ''' <param name="additionalSymbols">Additional field for property symbols initialized with the initializer.</param>
         ''' <param name="next">The next binder.</param>
         ''' <param name="root">Root syntax node</param>
-        Public Sub New(symbol As Symbol, [next] As Binder, root As VisualBasicSyntaxNode)
+        Public Sub New(symbol As Symbol, additionalSymbols As ImmutableArray(Of Symbol), [next] As Binder, root As VisualBasicSyntaxNode)
             MyBase.New([next])
             Debug.Assert((symbol.Kind = SymbolKind.Field) OrElse (symbol.Kind = SymbolKind.Property) OrElse (symbol.Kind = SymbolKind.Parameter))
+            Debug.Assert(additionalSymbols.All(Function(s) s.Kind = SymbolKind.Field OrElse s.Kind = SymbolKind.Property))
```

Comment:
>additionalSymbols.All(Function(s) s.Kind = SymbolKind.Field OrElse s.Kind = SymbolKind.Property) [](start = 25, length = 96)

It feels like the assert should be stronger - 
- an empty set for symbol.Kind = SymbolKind.Parameter
- s.Kind = symbol.Kind